apps/dashboard/src/components/auth/inbox-playground.tsx

```diff
@@ -0,0 +1,349 @@
+import { useAuth } from '../../context/auth/hooks';
+import { getComponentByType } from '../workflow-editor/steps/component-utils';
+import { StepTypeEnum, UiComponentEnum, WorkflowCreationSourceEnum } from '@novu/shared';
+import { createWorkflow } from '../../api/workflows';
+import { useWorkflows } from '../../hooks/use-workflows';
+import { FormProvider, useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { z } from 'zod';
+import { showToast } from '../primitives/sonner-helpers';
+import { ToastIcon } from '../primitives/sonner';
+import { ROUTES } from '../../utils/routes';
+import { useNavigate } from 'react-router-dom';
+import { Popover, PopoverContent, PopoverTrigger } from '../primitives/popover';
+import { InlineToast } from '../primitives/inline-toast';
+
+interface PreviewStyle {
+  id: string;
+  label: string;
+  image: string;
+}
+
+const previewStyles: PreviewStyle[] = [
+  { id: 'popover', label: 'Popover', image: '/images/auth/popover-layout.svg' },
+  { id: 'sidebar', label: 'Side Menu', image: '/images/auth/sidebar-layout.svg' },
+  { id: 'full-width', label: 'Full Width', image: '/images/auth/full-width-layout.svg' },
+];
+
+export function InboxPlayground() {
+  const form = useForm({
+    mode: 'onSubmit',
+    resolver: zodResolver(
+      z.object({
+        subject: z.string(),
+        body: z.string(),
+        primaryColor: z.string(),
+        foregroundColor: z.string(),
+        primaryAction: z.object({
+          label: z.string(),
+          redirect: z.object({
+            target: z.string(),
+            url: z.string(),
+          }),
+        }),
+        secondaryAction: z.object({
+          label: z.string(),
+          redirect: z.object({
+            target: z.string(),
+            url: z.string(),
+          }),
+        }),
+      })
+    ),
+    values: {
+      subject: '**Welcome to Inbox!**',
+      body: 'This is your first notification. Customize and explore more features.',
+      primaryColor: '#DD2450',
+      foregroundColor: '#0E121B',
+      primaryAction: {
+        label: 'Add to your app',
+        redirect: {
+          target: '_self',
+          url: '/',
+        },
+      },
+      secondaryAction: null,
+    },
+    shouldFocusError: true,
+  });
+
+  const [selectedStyle, setSelectedStyle] = useState<string>('popover');
+  const [openAccordion, setOpenAccordion] = useState<string | undefined>('layout');
+  const { triggerWorkflow, isPending } = useTriggerWorkflow();
+  const { data } = useWorkflows();
+  const auth = useAuth();
+  const [hasNotificationBeenSent, setHasNotificationBeenSent] = useState(false);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (!data) return;
+
+    (async () => {
+      const workflow = data?.workflows.find((workflow) => workflow.workflowId === 'onboarding-demo-workflow');
+
+      if (!workflow) {
+        await createWorkflow({
```

Comment:
Here we are creating a new onboarding workflow that will be used to trigger the inbox demo, and user implementation. We only create it once.